src/filemanager/filemanager.cpp

```diff
@@ -1,26 +1,31 @@
 #include "filemanager.h"
 
 #include <QTextStream>
-#include <QException> // temp
 #include <QFile>
 
-QString FileManager::readFromFile(const QString& fname)
+#include "utils.h"
+
+QString FileManager::readFromFile(const QString &fileName)
 {
-    QFile file(fname);
+    if (fileName.isEmpty())
+    {
+        throw IncorrectUserInput();
+    }
+    QFile file(fileName);
 
     if (file.open(QIODevice::ReadOnly))
     {
         QTextStream stream(&file);
-        QString readResult = stream.readAll();
+        QString rReadResult = stream.readAll();
         file.close();
-        return readResult;
+        return rReadResult;
     }
     throw QException();
```

Comment:
maybe better idea is to create your own exception

Reply:
done